### PR TITLE
Graceful shutdown

### LINKED ISSUE #40: Allow usage of NettyEventLoops
Currently it is not possible to provide com.aerospike.client.async.NettyEventLoop instance when creating new client instance, as internally it uses ^com.aerospike.client.async.NioEventLoop meta hints on bindings.
I believe those meta hints should be replaced with more generic com.aerospike.client.async.EventLoops interface in order to allow alternative implementations.

### LINKED ISSUE #41: Unconditional eventLoops.close() call when stopping client
When closing an aerospike client in `stop-aerospike-client` function, corresponding EventLoops instance is being closed unconditionally, which causes problems when having multiple clients with shared EventLoops instance.
In this case, the lib attempts to close same instance multiple times, which leads to hangs.
I propose to close the client's EventLoops instance only in case it was created automatically by the library, and not closed when an external instance is provided via `:event-loops` configuration parameter - in this case its lifecycle must be managed by a user.

### PR DESCRIPTION
**Pull request by sobersanta**
Conditional EventLoops close on client stop
(closes #41 )

Usage of EventLoops vs NioEventLoops interface where applicable
(closes #40 )

### CODE REVIEW

File: src/aerospike_clj/client.clj [hunk 1]
```diff
@@ -63,27 +64,31 @@
   \"username\"
   :port (default is 3000)
   :client-events an implementation of ClientEvents. Either a single one or a vector
-  thereof. In the case of a vector, the client will chain the instances by order."
+  thereof. In the case of a vector, the client will chain the instances by order.
+  If no event-loops instance is provided, single-threaded one is created automatically."
```
Comment (barkanido) [inline src/aerospike_clj/client.clj]: provided,  **a** single-threaded

File: src/aerospike_clj/client.clj [hunk 2]
```diff
@@ -63,27 +64,31 @@
...
   ([hosts aero-ns]
    (init-simple-aerospike-client hosts aero-ns {}))
   ([hosts aero-ns conf]
    (let [cluster-name (utils/cluster-name hosts)
-         event-loops (:event-loops conf (create-event-loops conf))
+         event-loops (:event-loops conf)
+         close-event-loops? (nil? event-loops)
```
Comment (barkanido) [inline src/aerospike_clj/client.clj]: maybe this should default to `true` and can be passed `{:close-event-loops? false}` to explicitly close it?
this couples the event-loops creation to its auto-closing.

**Review by barkanido — COMMENTED** [review]
aside for a small comment LGTM.

File: src/aerospike_clj/client.clj (hunk 2 above)
Comment (barkanido) [reply src/aerospike_clj/client.clj]: OTOH, the creator should destroy it as a good thumb rule.

Comment (sobersanta) [reply src/aerospike_clj/client.clj]: Discussed in offline, keeping as is

File: src/aerospike_clj/client.clj (hunk 1 above)
Comment (sobersanta) [reply src/aerospike_clj/client.clj]: Updated the docs, please review and if everything is fine, please resolve.

**Review by barkanido — APPROVED** [review]

**Pull request merged**